core/dbt/graph/selector_spec.py

```diff
@@ -121,7 +122,7 @@ def selection_criteria_from_dict(
 
         # If defined field in selector, override CLI flag
         indirect_selection = IndirectSelection(
-            dct.get("indirect_selection", None) or indirect_selection
+            dct.get("indirect_selection", get_flags().INDIRECT_SELECTION)
```

Comment:
This is the most important change that leads to resolving the bug and also removes all of the variable pass-through.

Reply:
What is the motivation for removing the variable pass-through here? I think it could be preferable to call `get_flags().INDIRECT_SELECTION` in `runnable.py` to avoid lower-level methods accessing global state.

Reply:
I looked up all call paths using this variable and all of them are referring to `flags. INDIRECT_SELECTION`. We can either pass it all the way everywhere or call it once in this lower level function(since this is where the selector being created.).
I think calling once at the lower level is a cleaner implementation overall. Since dbt is built on the assumption of flags is the first thing being constructed during an invocation.

Reply:
Okay, that's fair. I recall from our 'de-globalization' conversations in the past that `flags` was more acceptable to access because its never mutated and initialized at the very beginning of an invocation, as you mentioned.